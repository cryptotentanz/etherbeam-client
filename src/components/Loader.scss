@use '../styles/resources/colors';
@use '../styles/resources/sizes';

// Variables

$color: colors.$dark-active;
$color-highlight: colors.$primary;

// Drawings

.my-loader {
  &-spinner,
  &-dots {
    width: sizes.$size;
    height: sizes.$size;
    margin: auto;
  }

  &-spinner {
    border: 4px solid $color;
    border-top: 4px solid $color-highlight;
    border-radius: 50%;
    animation: spin 2s linear infinite;
  }

  &-dots {
    position: relative;
    width: 10px;
    height: 10px;
    border-radius: 5px;
    background-color: $color;
    color: $color;
    animation: flash 1s infinite linear alternate;
    animation-delay: 0.5s;

    &:before,
    &:after {
      content: '';
      display: inline-block;
      position: absolute;
      top: 0;
      width: 10px;
      height: 10px;
      border-radius: 5px;
      background-color: $color;
      color: $color;
      animation: flash 1s infinite alternate;
    }

    &:before {
      left: -15px;
      animation-delay: 0s;
    }

    &:after {
      left: 15px;
      animation-delay: 1s;
    }
  }
}

// Animations

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes flash {
  0% {
    background-color: $color-highlight;
  }
  50%,
  100% {
    background-color: $color;
  }
}
