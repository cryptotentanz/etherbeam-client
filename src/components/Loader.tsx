import React, { FC } from 'react'
import './Loader.scss'

type LoaderTypeProp = 'spinner' | 'dots'

interface LoaderProps {
  type: LoaderTypeProp
}

const Loader: FC<LoaderProps> = ({ type }: LoaderProps) => {
  return (
    <div className="my-loader">
      <div className={`my-loader-${type}`}></div>
    </div>
  )
}

export default Loader
