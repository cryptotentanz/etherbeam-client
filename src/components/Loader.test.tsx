import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Loader from './Loader'

describe('<Loader />', () => {
  let component: ShallowWrapper

  describe('Spinner', () => {
    beforeAll(() => {
      component = shallow(<Loader type="spinner" />)
    })

    it('should render', () => {
      expect(component.find('div').at(0).hasClass('my-loader')).toBeTruthy()
      expect(component.find('div').at(0).children().find('div').at(0).hasClass('my-loader-spinner')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Dots', () => {
    beforeAll(() => {
      component = shallow(<Loader type="dots" />)
    })

    it('should render', () => {
      expect(component.find('div').at(0).hasClass('my-loader')).toBeTruthy()
      expect(component.find('div').at(0).children().find('div').at(0).hasClass('my-loader-dots')).toBeTruthy()
    })
  })
})
