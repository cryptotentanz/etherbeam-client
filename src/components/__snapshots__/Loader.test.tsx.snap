// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Spinner should match snapshot 1`] = `
<div
  class="my-loader"
>
  <div
    class="my-loader-spinner"
  />
</div>
`;
